Core/Object Arts/Dolphin/Database/DBColAttr.cls

```diff
@@ -199,40 +203,39 @@ specialFlagAt: flagMask put: aBoolean
 
 	special mask: flagMask set: aBoolean!
 
-type
-	"Answer the sqlType instance variable."
-
-	^sqlType!
-
-type: anInteger
-	"Private - Set the sqlType instance variable to anInteger."
-
-	sqlType := anInteger!
+transferOctetLength
 
-typeLength
+	"Private - Return the transfer octet length (size in bytes of the buffer necessary to receive data for this column) based on the column type.
+	https://docs.microsoft.com/en-us/sql/odbc/reference/appendixes/transfer-octet-length"
 
-	"Private - Return the length (size in bytes) of the receiver based on its type."
-
-	| cType |
-
-	cType := self cType.
-
-	(cType = SQL_C_BIT or: [cType = SQL_C_TINYINT]) ifTrue: [^1].
-	cType = SQL_C_SHORT ifTrue: [^2].
-	(cType = SQL_C_LONG or: [cType = SQL_C_FLOAT]) ifTrue: [^4].
-	(cType = SQL_C_DATE or: [cType = SQL_C_TIME]) ifTrue: [^6].
-	cType = SQL_C_DOUBLE ifTrue: [^8].
-	(cType = SQL_C_TIMESTAMP or: [cType = SQL_C_GUID]) ifTrue: [^16].
+	(sqlType = SQL_BIT or: [sqlType = SQL_TINYINT]) ifTrue: [^1].
+	sqlType = SQL_SMALLINT ifTrue: [^2].
+	(sqlType = SQL_INTEGER or: [sqlType = SQL_REAL]) ifTrue: [^4].
+	(sqlType = SQL_TYPE_DATE or: [sqlType = SQL_TYPE_TIME]) ifTrue: [^6].
+	(sqlType = SQL_FLOAT or: [sqlType = SQL_DOUBLE]) ifTrue: [^8].
+	(sqlType = SQL_TYPE_TIMESTAMP or: [sqlType = SQL_GUID]) ifTrue: [^16].
 
 	"SQL_BIGINT
 	The number of bytes required to hold the character representation of this data if the character set is ANSI, and twice this number if the character set is UNICODE, because this data type is returned as a character string by default. The character representation consists of 20 characters: 19 for digits and a sign, if signed, or 20 digits, if unsigned. Therefore, the length is 20."
-	self type = SQL_BIGINT ifTrue: [^##(20 * 2 "allow for unicode")].
+	sqlType = SQL_BIGINT ifTrue: [^##(20 * 2 "allow for unicode")].
 
 	"SQL_DECIMAL, SQL_NUMERIC
 	The number of bytes required to hold the character representation of this data if the character set is ANSI, and twice this number if the character set is UNICODE. This is the maximum number of digits plus two, because the data is returned as a character string and characters are needed for the digits, a sign, and a decimal point. For example, the transfer length of a column defined as NUMERIC(10,3) is 12."
-	(self type = SQL_DECIMAL or: [self type = SQL_NUMERIC]) ifTrue: [^self length: (self precision + 2) * 2 "allow for unicode"].
+	(sqlType = SQL_DECIMAL or: [sqlType = SQL_NUMERIC]) ifTrue: [^(self precision + 2) * 2 "allow for unicode"].
```

Comment:
I think it would be find to just simplify this away and return the largest possible size, then you wouldn't need the switch :-). The maximum size is relatively small and the data will need to be transferred as a string and parsed anyway, so the benefit in being precise is small. It's not worth spending more time on though, and good to bottom it out. 
It makes me think it would be useful to allocate some slop around byte objects in the debug VM and fill it with a pattern that could be checked for overwrites on GC. Getting the size of these buffers wrong has unpleasant effects as all this case has demonstrated.